### PR TITLE
Feature/add node customisation ui

### PR DESCRIPTION
**Pull request by ThomasCrund**
Added Customisation Ability To Nodes and implemented ui changes for
- Removed Rename Option
- Put Inputs Above Outputs
- Added Dragging over entire node not just title bar
- Added a X in the corner to delete nodes

### CODE REVIEW

**Comment by 132ikl** [comment]
You can't drag on the inputs/outputs section of the node, is there any way to fix this?

**Comment by ThomasCrund** [comment]
I'm not sure it is possible to get drag to work on the inputs and outputs unless you do customise the node interfaces's as well to emit when they are clicked on. I don't know how todo that or if its possible

**Comment by 132ikl** [comment]
Can you do that :flushed:

**Review by 132ikl — APPROVED** [review]

File: src/components/node-editor/NodeOpsi.vue
Comment (carbotaniuman) [inline src/components/node-editor/NodeOpsi.vue:113]: If any of these are the same as `Components.Node`, can we let inheritance specify them and omit them from the file?

**Review by carbotaniuman — CHANGES_REQUESTED** [review]

File: src/components/node-editor/NodeOpsi.vue
Comment (ThomasCrund) [reply src/components/node-editor/NodeOpsi.vue:113]: Did i fix the issue with code that was not needed being in there

**Review by carbotaniuman — APPROVED** [review]

**Review by 132ikl — CHANGES_REQUESTED** [review]
here's a couple more minor ui tweaks i think we should make before merging (if you want to do it on a separate branch it's fine though, let me know and i will approve + merge):
1. fix this issue (two menus open at once): https://fs.ikl.sh/selif/inkzlu6w.webm
2. fix this issue (dragging with menu open): https://fs.ikl.sh/selif/onyjgddp.webm
3. fix this issue (draggin with menu open, slightly different): https://fs.ikl.sh/selif/z3w628wm.webm

**Review by 132ikl — CHANGES_REQUESTED** [review]
If while dragging the nodetree window your cursor runs into a node, the dragging is stopped. Ready to merge after this fix.

https://fs.ikl.sh/selif/id9xa5ry.webm

**Pull request merged**
